src/nvim/window.c

```diff
@@ -2451,13 +2450,18 @@ int win_close(win_T *win, bool free_buf)
     EMSG(_("E814: Cannot close window, only autocmd window would remain"));
     return FAIL;
   }
-  if ((firstwin == win && lastwin_nofloating() == win)
-      && lastwin->w_floating) {
-    // TODO(bfredl): we might close the float also instead
+  if (win->w_floating && win->w_float_config.relative == kFloatRelativeWindow
```

Comment:
what is this for? closing the float _itself_ should never be a problem.

Reply:
I used it for the case when the float that is being closed is itself related to the last non-floating window. We shouldn't close the float in that case.
The only other way that I could think of was to check that a window was not the last non-floating every time `win_close` was called, but that would mean that `win_close` won't be returning `FAIL` for the case, as the check would have happened before it was called. (I thought keeping all the checks in a single function would be neater :sweat_smile:.)
Should I change it to the latter option? Or do you have something else in mind?

Reply:
> I used it for the case when the float that is being closed is itself related to the last non-floating window. We shouldn't close the float in that case.

But it is not correct. There is no reason to stop closing a _floating window_ just because it is anchored. How would a plugin managing such a float close it? should it un-anchor the float before closing it? that would just be byzantine.

Reply:
Ah, right. I totally missed that point.
A question:
If a user tries to close the last non floating window, should the floating windows anchored to it be closed before displaying an error message? Or should they be left open?

Reply:
in case closing the main window fails, the float should not be closed.